backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Core/Services/LookupService.cs

```diff
@@ -32,8 +35,22 @@ public async Task<Party> GetPartyForOrganization(string organizationNumber)
         /// <inheritdoc/>        
         public async Task<Party> GetPartyFromReporteeListIfExists(int partyId)
         {
-            Party partyInfo = await _lookupClient.GetPartyFromReporteeListIfExists(partyId);
+            Party partyInfo = await _lookupClient.GetPartyFromReporteeListIfExists(partyId);    
             return partyInfo;
         }
+
+        /// <inheritdoc/>        
+        public async Task<Party> GetPartyByUUID(Guid uuid)
+        {
+            List<Party> partyList = await _registerClient.GetPartyListByUUID(new List<Guid>() { uuid });
```

Comment:
Yeah, I mostly just followed the already established pattern. Though, in defence of this approach, it is kinda nice that the frontend doesn't have to know that parties are fetched from Registry and users are fetched from UserProfile - it's all abstracted away in the BFF and then frontend can have one single and clean rtkqAPI file for all the lookup stuff.

Ofc, this isn't nessesarily the best solution, but I think it's something to be discussed (and possibly changed) in the same process as the existence of the LookupController as a whole

Reply:
Yeah I agree that the pros of this might be bigger than the cons of this approach(where the cons really just is about it looking weird). Also it makes sense having business logic in the service class.

Reply:
> Though, in defence of this approach, it is kinda nice that the frontend doesn't have to know that parties are fetched from Registry and users are fetched from UserProfile - it's all abstracted away in the BFF and then frontend can have one single and clean rtkqAPI file for all the lookup stuff.

No I changed my mind. Now after looking more detailed in LookupController, I don't understand the pros of doing it that way. But the real problem here is really our controller names, some controllers are more general like lookupController and some are less. I think there's something wrong with our controller names when we can debate putting them in another controller. E.g. LookupController and ResourceController are too general and it can always be debated if one endpoint should be in one or the other (and it's the same with the other controllers as well e.g. MaskinportenschemaController instead of ResourceController).

With that being said what do you think about having a workshop about this and find what the different controllers should be named, what they should contain and if we should remove some of them? As I think this problem will come up again and again.

Reply:
I think a workshop is a great idea.
We need to go through the controllers and find a system that we can make our own before it becomes too big of a problem. And since things are slow now anyway, the timing is good.